src/parsers/pm2ps_parser.jl

```diff
@@ -263,41 +263,76 @@ function make_thermal_gen(gen_name, d, bus)
     return thermal_gen
 end
 
-function read_gen(data,Buses)
+"""
+Transfer generators to ps_dict according to their classification
+"""
+function read_gen(data, Buses, genmap_file=nothing)
+
+    if genmap_file == nothing
+        genmap_file = joinpath(dirname(dirname(pathof(PowerSystems))),
+                               "src/parsers/generator_mapping.toml")
+    end
+    genmap_dict = TOML.parsefile(genmap_file) # not working... map to current path??
+    
     Generators = Dict{String,Any}()
     Generators["Thermal"] = Dict{String,Any}()
     Generators["Hydro"] = Dict{String,Any}()
     Generators["Renewable"] = Dict{String,Any}()
     Generators["Renewable"]["PV"]= Dict{String,Any}()
     Generators["Renewable"]["RTPV"]= Dict{String,Any}()
     Generators["Renewable"]["WIND"]= Dict{String,Any}()
-    Generators["Storage"] = Dict{String,Any}()
+    Generators["Storage"] = Dict{String,Any}() # not currently used? JJS 3/13/19
+    
     if haskey(data,"gen")
```

Comment:
This is just a style thing (and I know you are just refactoring existing code), but now you have a long section of code in an if statement where the else clause just returns.  I would check for the not present case first and return early.  Then you eliminate the indentation in the rest of the code.

Reply:
Good observation. I made this change. Now that I think about it, the if statement could go right at the top of the function